src/data/polynomial/partial_fractions.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2022 The Xena Project. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Sidharth Hariharan
+-/
+import ring_theory.localization.fraction_ring -- field of fractions
+import data.polynomial.div -- theory of division and remainder for monic polynomials
+import tactic.field_simp
+import tactic.linear_combination
+import data.zmod.basic
+import logic.function.basic
+
+/-!
+
+# Partial fractions
+
+These results were formalised by the Xena Project, at the suggestion
+of Patrick Massot.
+
+
+## The main theorems
+
+* General partial fraction decomposition theorem for polynomials over an integral domain R :
+  if f, g₁, g₂, ..., gₙ ∈ R[X] and the gᵢs are all monic and pairwise coprime, then ∃ q, r₁, ..., rₙ
+  ∈ R[X] such that f / g₁g₂...gₙ = q + r₁/g₁ + ... + rₙ/gₙ and for all i, deg(rᵢ) < deg(gᵢ).#check
+
+* The result is formalized here in slightly more generality, using finsets.
+
+* The proof is done by proving the two-denominator case and then performing finset induction for an
+  arbitrary (finite) number of denominators.
+
+## Scope for Expansion
+
+* Proving uniqueness of the decomposition
+
+-/
+
+
+-- Let `R` be an integral domain
+variables (R : Type) [comm_ring R] [is_domain R]
+
+open_locale polynomial
+
+open polynomial
+
+-- Let K be the field of fractions of R[X].
+variables (K : Type) [field K] [algebra R[X] K]  [is_fraction_ring R[X] K]
+
+section one_denominator
+
+namespace polynomial
+
+variables (f : R[X]) {g : R[X]}
+
+lemma div_eq_quo_add_rem_div (hg : g.monic) : ∃ q r : R[X], r.degree < g.degree ∧
```

Comment:
This can maybe be moved to a more basic file, but I am not sure there is a reasonable one.